code/modules/mob/living/carbon/life.dm

```diff
@@ -384,3 +383,39 @@
 		if(360.15 to INFINITY) //360.15 is 310.15 + 50, the temperature where you start to feel effects.
 			//We totally need a sweat system cause it totally makes sense...~
 			bodytemperature += min((body_temperature_difference / BODYTEMP_AUTORECOVERY_DIVISOR), -BODYTEMP_AUTORECOVERY_MINIMUM)	//We're dealing with negative numbers
+/////////
+//LIVER//
+/////////
+
+/mob/living/carbon/proc/handle_liver()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(((!(NOLIVER in dna.species.species_traits)) && (!liver)) || liver && liver.damage >= 100)
+		if(liver)
+			liver.failing = TRUE

```

Comment:
This is just a mess now. NOLIVER species with a damaged liver will still experience liver failure. I suggest branching if(liver) immediately after defining it, then doing all the other logic inside the branches.